pkg/tfgen/installation_docs.go

```diff
@@ -401,14 +401,16 @@ func getDefaultHeadersToSkip() []*regexp.Regexp {
 		regexp.MustCompile("[Dd]ebugging"),
 		regexp.MustCompile("[Tt]erraform CLI"),
 		regexp.MustCompile("[Tt]erraform Cloud"),
+		regexp.MustCompile("Delete Protection"),
 	}
 	return defaultHeaderSkipRegexps
 }
 
 func getTfVersionsToRemove() []*regexp.Regexp {
 	tfVersionsToRemove := []*regexp.Regexp{
-		regexp.MustCompile(`It requires [tT]erraform [v0-9]+\.?[0-9]?\.?[0-9]? or later.`),
+		regexp.MustCompile(`(It|This provider) requires(.*|at least) [tT]erraform [v0-9]+\.?[0-9]?\.?[0-9]?(.*|or later).`),
```

Comment:
```suggestion
		regexp.MustCompile(`(It|This provider) requires( at least)? [tT]erraform [v0-9]+\.?[0-9]?\.?[0-9]?( or later)?.`),
```

I think that `(.*|at least)` is match anything or "at least", which I think is equivalent to `.*`. I suspect the more specific clause never triggers, since the spacing is wrong.